### PR TITLE
pAIs Play Music

### PR DESCRIPTION
**Pull request by hubismal**
## About the PR 
PAIs can play MIDIs on their own nokia-esque synths accessible through the action menu.
The spacestation14.sf2 soundfont has a new program on bank 1 called "paiSynth" that uses a cell phone beep sample, as there wasn't a consistently chiptuney midi instrument, at least in my default soundfont.
The pAI prototype now has an instrument that isn't accessible through interaction, and is instead opened up by the PAIMidi action that PAIs start out with.
Instruments now can also accept players that are the instrument owner itself.
In the future I'd like to reduce the range on these instruments, but I've been advised by Vera that to do this right now with the midi system would be somewhat complicated.

This is my first PR. Prepare for some facepalming code wise.

**Screenshots**
The only visual embellishment is the icon for the action.
image

**Changelog**
:cl:
- add: pAIs have been outfitted with cheap ringtone synthesizers to play their own music.

### CODE REVIEW

File: BuildChecker/BuildChecker.csproj [hunk 1]
```diff
@@ -33,7 +33,9 @@ https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild
     <OutputPath>bin\Release\</OutputPath>
   </PropertyGroup>
   <Target Name="Build">
+		<!--
```
Comment (20kdc) [inline BuildChecker/BuildChecker.csproj:36]: uuuuuuh

File: Resources/Prototypes/Actions/actions.yml [hunk 2]
```diff
@@ -66,6 +66,20 @@
...
+  actionType: PAIMidi
+  icon: Interface/Actions/midi.png
+  name: "Play MIDI"
+  description: "Open your portable MIDI interface to soothe your owner."
+  filters:
+    - pai
+    - music
+  behaviorType: Instant
+  behavior: !type:PAIMidi
+    radius: 3
```
Comment (20kdc) [inline Resources/Prototypes/Actions/actions.yml:79]: This is an instant action, so it doesn't have these three fields, I think?

File: Resources/Prototypes/Entities/Objects/Fun/vehicle.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+- type: entity
```
Comment (20kdc) [inline Resources/Prototypes/Entities/Objects/Fun/vehicle.yml:1]: Not sure what this is doing here

**Comment by 20kdc** [comment]
would like to mention that you left the `paiSynth` sample (unused) in the SS14 soundfont file

**Comment by hubismal** [comment]
Okay somehow this got mangled with an old, entirely separate branch i haven't touched for months. Let me remove all of this garbage.

File: Content.Server/Actions/Actions/PAIMidi.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    {
+
+        public void DoInstantAction(InstantActionEventArgs args)
+        {
+            if (!args.Performer.TryGetComponent<ServerUserInterfaceComponent>(out var serverUi)) return;
+            if (!args.Performer.TryGetComponent<ActorComponent>(out var actor)) return;
+			if (!serverUi.TryGetBoundUserInterface(InstrumentUiKey.Key,out var bui)) return;
+
+			bui.Toggle(actor.PlayerSession);
+
```
Comment (mirrorcult) [inline Content.Server/Actions/Actions/PAIMidi.cs:31]: whitespace is fucked up here

File: Content.Server/PAI/PAISystem.cs [hunk 5]
```diff
@@ -97,6 +101,15 @@ private void OnMindAdded(EntityUid uid, PAIComponent pai, MindAddedMessage args)
         private void PAITurningOff(EntityUid uid)
         {
             UpdatePAIAppearance(uid, PAIStatus.Off);
+            //  Close the instrument interface if it was open
+            //  before closing
+            if (EntityManager.TryGetComponent<ServerUserInterfaceComponent>(uid, out var serverUi))
+            if (EntityManager.TryGetComponent<ActorComponent>(uid, out var actor))
+            if (serverUi.TryGetBoundUserInterface(InstrumentUiKey.Key,out var bui))
+			bui.Close(actor.PlayerSession);
```
Comment (mirrorcult) [inline Content.Server/PAI/PAISystem.cs:109]: whitespace again

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/PAI/PAISystem.cs [hunk 6]
```diff
@@ -97,6 +101,15 @@ private void OnMindAdded(EntityUid uid, PAIComponent pai, MindAddedMessage args)
         private void PAITurningOff(EntityUid uid)
         {
             UpdatePAIAppearance(uid, PAIStatus.Off);
+            //  Close the instrument interface if it was open
+            //  before closing
+            if (EntityManager.TryGetComponent<ServerUserInterfaceComponent>(uid, out var serverUi))
+            if (EntityManager.TryGetComponent<ActorComponent>(uid, out var actor))
+            if (serverUi.TryGetBoundUserInterface(InstrumentUiKey.Key,out var bui))
+            bui.Close(actor.PlayerSession);
```
Comment (mirrorcult) [inline Content.Server/PAI/PAISystem.cs:109]: why is there just no indentation here

Comment (hubismal) [reply Content.Server/PAI/PAISystem.cs]: editor indentation settings apparently

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**